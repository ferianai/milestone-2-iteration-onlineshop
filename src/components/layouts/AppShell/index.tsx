import Navbar from "../Navbar";
import { useRouter } from "next/router";

type AppShellProps = {
    children: React.ReactNode;
};

const disableNavbar = ["/auth/login", "/auth/register"];

const AppShell: React.FC<AppShellProps> = (props: AppShellProps ) => {
    const { children } = props;
    const { pathname } = useRouter();

    return (
        <main className='app-shell'>
            {!disableNavbar.includes(pathname) && <Navbar />}
            <div className='content'>
                {children}
            </div>
            {/* <div className='content'>Footer</div> */}
        </main>
    );
}

export default AppShell;