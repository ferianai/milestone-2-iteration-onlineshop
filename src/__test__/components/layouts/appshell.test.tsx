import AppShell from "@/components/layouts/AppShell";
import { render } from '@testing-library/react';

// Mock the useRouter hook
jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    pathname: '/',  
    query: {},
    push: jest.fn(),
    replace: jest.fn(),
  }),
}));

test('AppShell renders with children', () => {
  const appShell = render(<AppShell children={<div>Test Content</div>} />);
  expect(appShell).toMatchSnapshot();
});

test('AppShell renders correctly', () => {
  const appShell = render(<AppShell children={<div>Test Content</div>} />);
  expect(appShell).toMatchSnapshot();
});