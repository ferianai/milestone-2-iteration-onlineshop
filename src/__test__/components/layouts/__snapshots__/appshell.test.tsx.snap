// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppShell renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="app-shell"
      >
        <nav
          class="navbar bg-blue-500"
        >
          <ul>
            <li>
              <a
                href="/"
              >
                ShopSmart
              </a>
            </li>
            <li>
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li>
              <a
                href="/product"
              >
                Product
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                href="/cart"
              >
                <span>
                  <i
                    class="fas fa-shopping-cart"
                  />
                   
                </span>
              </a>
            </li>
            <li>
              <a
                class="font-semibold"
                href="/auth/login"
              >
                Login 
                <i
                  class="fa-solid fa-arrow-right"
                />
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="content"
        >
          <div>
            Test Content
          </div>
        </div>
        <div>
          <div
            class="bg-gray-800"
          >
            <div
              class="footer"
            >
              <div
                class="footerSection"
              >
                <h2
                  class="text-xl font-bold mb-2"
                >
                  ShopSmart
                </h2>
                <p>
                  Online Store: Shopping Made Simple, Smart & Affordable.
                </p>
              </div>
              <div
                class="footerSection"
              >
                <h3
                  class="text-lg font-semibold mb-2"
                >
                  Help
                </h3>
                <ul>
                  <li>
                    Telephone: 031-1-876-876
                  </li>
                  <li>
                    Email: customer.care@shopsmart.com
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/help"
                    >
                      Help Page
                    </a>
                  </li>
                  <li>
                    ShopSmartCare
                  </li>
                  <li>
                    Consumer Complaints Service of the Directorate General of Consumer Protection and Trade Order, Ministry of Trade of the Republic of Indonesia
                  </li>
                </ul>
              </div>
              <div
                class="footerSection"
              >
                <h3
                  class="text-lg font-semibold mb-2"
                >
                  ShopSmart Info
                </h3>
                <ul>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/about"
                    >
                      About ShopSmart
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/blog"
                    >
                      ShopSmart Friends Blog
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/press"
                    >
                      Press Release
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/news"
                    >
                      Latest News
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/careers"
                    >
                      Career
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/terms"
                    >
                      Terms & Privacy Policy
                    </a>
                  </li>
                </ul>
              </div>
              <div
                class="footerSection"
              >
                <h3
                  class="text-lg font-semibold mb-2"
                >
                  Security & Privacy
                </h3>
                <div
                  class="certificates"
                >
                  <img
                    alt="SSL Certificate"
                    class="pb-4"
                    src="https://www.static-src.com/siva/asset/11_2024/ISO-27001.jpg?w=100"
                  />
                  <img
                    alt="PCI DSS Certificate"
                    class="pb-4"
                    src="https://www.static-src.com/siva/asset/11_2024/ISO-27701.jpg?w=100"
                  />
                  <img
                    alt="GDPR Certificate"
                    class="pb-4"
                    src="https://www.static-src.com/siva/asset/11_2024/ceritpedia.jpg?w=100"
                  />
                </div>
              </div>
              <div
                class="footerSection"
              >
                <h3
                  class="text-lg font-semibold mb-2"
                >
                  Follow Us
                </h3>
                <p>
                  Stay connected with ShopSmart on social media!
                </p>
              </div>
              <button
                class="fixed bottom-4 right-4 bg-blue-300 text-white text-4xl p-4 rounded-md shadow-lg hover:bg-blue-600 transition"
                title="Back to Top"
              >
                ↑
              </button>
            </div>
            <div
              class="footerBottom text-center mt-4"
            >
              <p>
                ShopSmart © 2025 | Endless Choices, One Online Store. All Rights Reserved
              </p>
              <p>
                Created by:
                 
                <a
                  class="text-blue-600"
                  href="https://github.com/Ferianai"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  @Ferianai
                </a>
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="app-shell"
    >
      <nav
        class="navbar bg-blue-500"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              ShopSmart
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/product"
            >
              Product
            </a>
          </li>
        </ul>
        <ul>
          <li>
            <a
              href="/cart"
            >
              <span>
                <i
                  class="fas fa-shopping-cart"
                />
                 
              </span>
            </a>
          </li>
          <li>
            <a
              class="font-semibold"
              href="/auth/login"
            >
              Login 
              <i
                class="fa-solid fa-arrow-right"
              />
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="content"
      >
        <div>
          Test Content
        </div>
      </div>
      <div>
        <div
          class="bg-gray-800"
        >
          <div
            class="footer"
          >
            <div
              class="footerSection"
            >
              <h2
                class="text-xl font-bold mb-2"
              >
                ShopSmart
              </h2>
              <p>
                Online Store: Shopping Made Simple, Smart & Affordable.
              </p>
            </div>
            <div
              class="footerSection"
            >
              <h3
                class="text-lg font-semibold mb-2"
              >
                Help
              </h3>
              <ul>
                <li>
                  Telephone: 031-1-876-876
                </li>
                <li>
                  Email: customer.care@shopsmart.com
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/help"
                  >
                    Help Page
                  </a>
                </li>
                <li>
                  ShopSmartCare
                </li>
                <li>
                  Consumer Complaints Service of the Directorate General of Consumer Protection and Trade Order, Ministry of Trade of the Republic of Indonesia
                </li>
              </ul>
            </div>
            <div
              class="footerSection"
            >
              <h3
                class="text-lg font-semibold mb-2"
              >
                ShopSmart Info
              </h3>
              <ul>
                <li>
                  <a
                    class="text-blue-400"
                    href="/about"
                  >
                    About ShopSmart
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/blog"
                  >
                    ShopSmart Friends Blog
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/press"
                  >
                    Press Release
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/news"
                  >
                    Latest News
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/careers"
                  >
                    Career
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/terms"
                  >
                    Terms & Privacy Policy
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="footerSection"
            >
              <h3
                class="text-lg font-semibold mb-2"
              >
                Security & Privacy
              </h3>
              <div
                class="certificates"
              >
                <img
                  alt="SSL Certificate"
                  class="pb-4"
                  src="https://www.static-src.com/siva/asset/11_2024/ISO-27001.jpg?w=100"
                />
                <img
                  alt="PCI DSS Certificate"
                  class="pb-4"
                  src="https://www.static-src.com/siva/asset/11_2024/ISO-27701.jpg?w=100"
                />
                <img
                  alt="GDPR Certificate"
                  class="pb-4"
                  src="https://www.static-src.com/siva/asset/11_2024/ceritpedia.jpg?w=100"
                />
              </div>
            </div>
            <div
              class="footerSection"
            >
              <h3
                class="text-lg font-semibold mb-2"
              >
                Follow Us
              </h3>
              <p>
                Stay connected with ShopSmart on social media!
              </p>
            </div>
            <button
              class="fixed bottom-4 right-4 bg-blue-300 text-white text-4xl p-4 rounded-md shadow-lg hover:bg-blue-600 transition"
              title="Back to Top"
            >
              ↑
            </button>
          </div>
          <div
            class="footerBottom text-center mt-4"
          >
            <p>
              ShopSmart © 2025 | Endless Choices, One Online Store. All Rights Reserved
            </p>
            <p>
              Created by:
               
              <a
                class="text-blue-600"
                href="https://github.com/Ferianai"
                rel="noopener noreferrer"
                target="_blank"
              >
                @Ferianai
              </a>
            </p>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppShell renders with children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="app-shell"
      >
        <nav
          class="navbar bg-blue-500"
        >
          <ul>
            <li>
              <a
                href="/"
              >
                ShopSmart
              </a>
            </li>
            <li>
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li>
              <a
                href="/product"
              >
                Product
              </a>
            </li>
          </ul>
          <ul>
            <li>
              <a
                href="/cart"
              >
                <span>
                  <i
                    class="fas fa-shopping-cart"
                  />
                   
                </span>
              </a>
            </li>
            <li>
              <a
                class="font-semibold"
                href="/auth/login"
              >
                Login 
                <i
                  class="fa-solid fa-arrow-right"
                />
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="content"
        >
          <div>
            Test Content
          </div>
        </div>
        <div>
          <div
            class="bg-gray-800"
          >
            <div
              class="footer"
            >
              <div
                class="footerSection"
              >
                <h2
                  class="text-xl font-bold mb-2"
                >
                  ShopSmart
                </h2>
                <p>
                  Online Store: Shopping Made Simple, Smart & Affordable.
                </p>
              </div>
              <div
                class="footerSection"
              >
                <h3
                  class="text-lg font-semibold mb-2"
                >
                  Help
                </h3>
                <ul>
                  <li>
                    Telephone: 031-1-876-876
                  </li>
                  <li>
                    Email: customer.care@shopsmart.com
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/help"
                    >
                      Help Page
                    </a>
                  </li>
                  <li>
                    ShopSmartCare
                  </li>
                  <li>
                    Consumer Complaints Service of the Directorate General of Consumer Protection and Trade Order, Ministry of Trade of the Republic of Indonesia
                  </li>
                </ul>
              </div>
              <div
                class="footerSection"
              >
                <h3
                  class="text-lg font-semibold mb-2"
                >
                  ShopSmart Info
                </h3>
                <ul>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/about"
                    >
                      About ShopSmart
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/blog"
                    >
                      ShopSmart Friends Blog
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/press"
                    >
                      Press Release
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/news"
                    >
                      Latest News
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/careers"
                    >
                      Career
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-blue-400"
                      href="/terms"
                    >
                      Terms & Privacy Policy
                    </a>
                  </li>
                </ul>
              </div>
              <div
                class="footerSection"
              >
                <h3
                  class="text-lg font-semibold mb-2"
                >
                  Security & Privacy
                </h3>
                <div
                  class="certificates"
                >
                  <img
                    alt="SSL Certificate"
                    class="pb-4"
                    src="https://www.static-src.com/siva/asset/11_2024/ISO-27001.jpg?w=100"
                  />
                  <img
                    alt="PCI DSS Certificate"
                    class="pb-4"
                    src="https://www.static-src.com/siva/asset/11_2024/ISO-27701.jpg?w=100"
                  />
                  <img
                    alt="GDPR Certificate"
                    class="pb-4"
                    src="https://www.static-src.com/siva/asset/11_2024/ceritpedia.jpg?w=100"
                  />
                </div>
              </div>
              <div
                class="footerSection"
              >
                <h3
                  class="text-lg font-semibold mb-2"
                >
                  Follow Us
                </h3>
                <p>
                  Stay connected with ShopSmart on social media!
                </p>
              </div>
              <button
                class="fixed bottom-4 right-4 bg-blue-300 text-white text-4xl p-4 rounded-md shadow-lg hover:bg-blue-600 transition"
                title="Back to Top"
              >
                ↑
              </button>
            </div>
            <div
              class="footerBottom text-center mt-4"
            >
              <p>
                ShopSmart © 2025 | Endless Choices, One Online Store. All Rights Reserved
              </p>
              <p>
                Created by:
                 
                <a
                  class="text-blue-600"
                  href="https://github.com/Ferianai"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  @Ferianai
                </a>
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="app-shell"
    >
      <nav
        class="navbar bg-blue-500"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              ShopSmart
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/product"
            >
              Product
            </a>
          </li>
        </ul>
        <ul>
          <li>
            <a
              href="/cart"
            >
              <span>
                <i
                  class="fas fa-shopping-cart"
                />
                 
              </span>
            </a>
          </li>
          <li>
            <a
              class="font-semibold"
              href="/auth/login"
            >
              Login 
              <i
                class="fa-solid fa-arrow-right"
              />
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="content"
      >
        <div>
          Test Content
        </div>
      </div>
      <div>
        <div
          class="bg-gray-800"
        >
          <div
            class="footer"
          >
            <div
              class="footerSection"
            >
              <h2
                class="text-xl font-bold mb-2"
              >
                ShopSmart
              </h2>
              <p>
                Online Store: Shopping Made Simple, Smart & Affordable.
              </p>
            </div>
            <div
              class="footerSection"
            >
              <h3
                class="text-lg font-semibold mb-2"
              >
                Help
              </h3>
              <ul>
                <li>
                  Telephone: 031-1-876-876
                </li>
                <li>
                  Email: customer.care@shopsmart.com
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/help"
                  >
                    Help Page
                  </a>
                </li>
                <li>
                  ShopSmartCare
                </li>
                <li>
                  Consumer Complaints Service of the Directorate General of Consumer Protection and Trade Order, Ministry of Trade of the Republic of Indonesia
                </li>
              </ul>
            </div>
            <div
              class="footerSection"
            >
              <h3
                class="text-lg font-semibold mb-2"
              >
                ShopSmart Info
              </h3>
              <ul>
                <li>
                  <a
                    class="text-blue-400"
                    href="/about"
                  >
                    About ShopSmart
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/blog"
                  >
                    ShopSmart Friends Blog
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/press"
                  >
                    Press Release
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/news"
                  >
                    Latest News
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/careers"
                  >
                    Career
                  </a>
                </li>
                <li>
                  <a
                    class="text-blue-400"
                    href="/terms"
                  >
                    Terms & Privacy Policy
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="footerSection"
            >
              <h3
                class="text-lg font-semibold mb-2"
              >
                Security & Privacy
              </h3>
              <div
                class="certificates"
              >
                <img
                  alt="SSL Certificate"
                  class="pb-4"
                  src="https://www.static-src.com/siva/asset/11_2024/ISO-27001.jpg?w=100"
                />
                <img
                  alt="PCI DSS Certificate"
                  class="pb-4"
                  src="https://www.static-src.com/siva/asset/11_2024/ISO-27701.jpg?w=100"
                />
                <img
                  alt="GDPR Certificate"
                  class="pb-4"
                  src="https://www.static-src.com/siva/asset/11_2024/ceritpedia.jpg?w=100"
                />
              </div>
            </div>
            <div
              class="footerSection"
            >
              <h3
                class="text-lg font-semibold mb-2"
              >
                Follow Us
              </h3>
              <p>
                Stay connected with ShopSmart on social media!
              </p>
            </div>
            <button
              class="fixed bottom-4 right-4 bg-blue-300 text-white text-4xl p-4 rounded-md shadow-lg hover:bg-blue-600 transition"
              title="Back to Top"
            >
              ↑
            </button>
          </div>
          <div
            class="footerBottom text-center mt-4"
          >
            <p>
              ShopSmart © 2025 | Endless Choices, One Online Store. All Rights Reserved
            </p>
            <p>
              Created by:
               
              <a
                class="text-blue-600"
                href="https://github.com/Ferianai"
                rel="noopener noreferrer"
                target="_blank"
              >
                @Ferianai
              </a>
            </p>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
